### ISSUE TITLE
scroll wheel shouldn't change dest tag

### ISSUE THREAD

**Issue by jedmccaleb** [stellar/stellar-client#1119]
hi, Jed. Some feedback of the stellar wallet. The wallet would listen to the middle mouse key and change the input box value. However, I wish you can remove this feature in the Tag input box. Two users reported they sent STR with a wrong tag today when deposit STR. As most gateways would use the tag to identify the user. It is no meaning to change the tag once user input it.

oh, it is the mouse wheel, not the middle key. Anyway, the tag value should not be changed when user touch the wheel.

**Comment by irisli** [comment]
# If you don't read this write up, at least read the TL;DR; at the bottom

Okay, for my janitorial, I did some analysis on this to hopefully settle (and understand) this issue.

# Overall visual differences
On iOs, it shows the symbolic keyboard. Desktop, there is no difference (with native/physical keyboard at least)
## input `type=text`
![inputtypetext](https://cloud.githubusercontent.com/assets/5728307/5656442/3e76eb92-9691-11e4-9c44-64ca4855c832.png)

## input `type=number`
![inputtypenumber](https://cloud.githubusercontent.com/assets/5728307/5656441/3a5813d8-9691-11e4-95e4-ceb8a4530dcb.png)

----------------------

## Flaw with all three
- ☒ Using up/down arrow keys changes the numbers

## 1. input `type=text`
Desktop:
- ☑☑ No scrolling problems
- ☑☑ Can still scroll while focused in element

Mobile:
- ☒ Starts off by default on ABC keyboard (in English at least)

## 2. naked input `type=number`
Desktop:
- ☒☒☒ Huge scrolling problem that can and will cause some people to lose money
- ☒☒☒ Pressing up or down arrow buttons will change the form

Mobile:
- ☑ Starts off by default on ABC keyboard (in English at least)

## 3. input `type=number` with disabling mousescroll
Desktop:
- ☒ hacky and unable to scroll while focused in and mouse is hovered over the input field. This is a minor inconvenience though

Mobile:
- ☑ Starts off by default on ABC keyboard (in English at least). Not a huge improvement

----------------------

# Analysis
Solution 2 can already be eliminated since it is the problem we are trying to solve.

Using solution 1 (type=text) is the simplest and avoids the scrolling problem.

Using solution 3 gives mobile (iOs) users a minorly faster/better UX (no having to switch keyboards).

To implement solution 3, we will need to write one directive:
- Disable mouse scroll

For maximum security we will also need to write:
- Disable up down arrow key (to disable up/down arrow key on dt fields)

We should also pay attention to amount if we want to optimize for usability (with a little hacky sacrificing). However, it is low priority since clicking the `[123]`` button is pretty straight forward on mobile devices and many also have a row of numbers on top.

In the long term though, we will still want to fix these problems and should go with solution 3.

----------------------

# TL;DR;
In the short term, we need to revert back to `type=text` for mousewheel safety. (the mobile number keyboard is overrated)

To make things pretty (just a minor improvement paid for with a minor hack), we should do solution 3.

In the long term, we should at least write a directive to disable up/down arrow key.

**Comment by bartekn** [comment]
The other solution would be to leverage information in `navigation.userAgent` (or use a script like [detectmobilebrowsers.com](http://detectmobilebrowsers.com/)) to detect mobile browser and depending on the outcome of the test display `number` on mobile and `text` on desktop.

**Comment by ReWeR-IV** [comment]
Sorry going offtopic but you're facing a problem with a recovery code , I've been trying to recover my password using my recovery code and everytime I put new password It's showing me "Forbidden." I have over 1 million stellar In my account. Any solution regarding that ?

**Comment by bekkibolthouse** [comment]
@ReWeR-IV please write in to the [helpdesk](https://stellarorg.zendesk.com/hc/en-us) and we'll take it from there.
